go/ct/spc/coordination_test.go

```diff
@@ -0,0 +1,186 @@
+package spc
+
+import (
+	"math/rand"
+	"regexp"
+	"runtime"
+	"strings"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/Fantom-foundation/Tosca/go/ct/gen"
+	"github.com/Fantom-foundation/Tosca/go/ct/rlz"
+	"github.com/Fantom-foundation/Tosca/go/ct/st"
+)
+
+// Test Condition
+type testCondition struct {
+	fits      bool
+	numStates int
+}
+
+func condition(fitsRule bool, numStates int) rlz.Condition {
+	return &testCondition{fits: fitsRule, numStates: numStates}
+}
+
+func (e testCondition) Check(s *st.State) (bool, error) {
+	return e.fits, nil
+}
+
+func (e testCondition) Restrict(generator *gen.StateGenerator) {
+}
+
+// the test condition produces exactly one test state per rule
+func (e testCondition) EnumerateTestCases(generator *gen.StateGenerator, consume func(*gen.StateGenerator) rlz.ConsumerResult) rlz.ConsumerResult {
+	for i := 0; i < e.numStates; i++ {
+		g := generator.Clone()
+		res := consume(g)
+		if res == rlz.ConsumeAbort {
+			return rlz.ConsumeAbort
+		}
+	}
+	return rlz.ConsumeContinue
+}
+
+func (e testCondition) String() string {
+	return "Test Condition"
+}
+
+func TestCoordination_GenerateMultipleStatesPerRule(t *testing.T) {
+	numJobs := runtime.NumCPU()
+	seed := 0
+	var counter atomic.Int64
+	opFunction := func(state *st.State) rlz.ConsumerResult {
+		counter.Add(1)
+		return rlz.ConsumeContinue
+	}
+	printFunction := func(time time.Duration, rate float64, current int64) {}
+
+	statesPerRule := 15
+
+	//var condition testCondition
+	rules := []rlz.Rule{}
+	testRule := rlz.Rule{
+		Name:      "test_rule",
+		Condition: condition(true, statesPerRule),
+		Effect:    rlz.NoEffect(),
```

Comment:
nit: the `Name` and `Effect` could likely be omitted for this test since they are not relevant, right?